File: linear-search/linearSearch.py
import random

def linearSearch(arr, sn):
    checks = 0
    for i in range(len(arr)):
        checks += 1
        if arr[i] == sn:
            return i, checks
    return -1, checks

#Test algorithm with random array
arr = [random.randint(1, 25) for n in range(25)]
print(arr)
sn = random.randint(1, 25)
index, checks = linearSearch(arr, sn)
print(f"Search number: {sn} \nIndex: {index} \nLinear search: {checks} elements checked\n")

File: linear-search/linearSearch.cpp
#include <iostream>
#include <random>
#include <vector>

using namespace std;

int* linear_search(int array[], int size, int sn){
    int checks = 0;
    static int result[2];

    for (int i = 0; i < size; i++){
        checks++;
        if (array[i] == sn){
            result[0] = i;
            result[1] = checks;
            return result;
        }
    }
    result[0] = -1;
    result[1] = checks;
    return result;
}

//Test algorithm with random array
int main() {
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> distrib(1, 25);
    int arr[25];
    for (int j = 0; j < 25; j++){
        arr[j] = distrib(gen);
    }
    cout << "[";
    for (int j = 0; j < 25; j++){
        cout << arr[j];
        if (j < 24) cout << ", ";
    }
    cout << "]" << endl;
    int sn = distrib(gen);
    int* result = linear_search(arr, 25, sn);
    int index = result[0];
    int checks = result[1];
    cout << "Search number: " << sn << endl;
    cout << "Index: " << index << endl;
    cout << "Linear search: " << checks << " elements checked" << endl << endl;
    return 0;
}